### PR TITLE
[Merged by Bors] - feat: port changes to `order/game_add`

### PR DESCRIPTION
**Pull request by vihdzp**
Simultaneously ports various PRs on this file.

This also fixes a mis-ported alias in `Mathlib/Order/RelIso/Basic.lean`

---

* [`order.game_add`@`6f870fad180eb0f5d8f1e4f385216c25513cb2f8`..`fee218fb033b2fd390c447f8be27754bc9093be9`](https://leanprover-community.github.io/mathlib-port-status/file/order/game_add?range=6f870fad180eb0f5d8f1e4f385216c25513cb2f8..fee218fb033b2fd390c447f8be27754bc9093be9)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Can you update the SHAs to reflect which commits you ported? You can use https://leanprover-community.github.io/mathlib-port-status/file/order/game_add to help.

File: Mathlib/Order/GameAdd.lean [hunk 1]
```diff
@@ -4,10 +4,11 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
 
 ! This file was ported from Lean 3 source module order.game_add
-! leanprover-community/mathlib commit 99e8971dc62f1f7ecf693d75e75fbbabd55849de
+! leanprover-community/mathlib commit fee218fb033b2fd390c447f8be27754bc9093be9
 ! Please do not edit these lines, except to modify the commit id
 ! if you have ported upstream changes.
 -/
+import Mathlib.Data.Sym.Sym2
 import Mathlib.Logic.Relation
 import Mathlib.Order.Basic
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:13]: Mathport output deletes this line, see https://leanprover-community.github.io/mathlib-port-status/file/order/game_add?range=99e8971dc62f1f7ecf693d75e75fbbabd55849de..fee218fb033b2fd390c447f8be27754bc9093be9

File: Mathlib/Order/GameAdd.lean [hunk 2]
```diff
@@ -19,16 +20,17 @@ This file defines, given relations `rα : α → α → Prop` and `rβ : β →
...
   decreases at a time.
 
-## Todo
-
-- Add custom `induction` and `fix` lemmas.
-- Define `Sym2.GameAdd`.
+- `Sym2.GameAdd`: the game addition relation on unordered pairs.
+- `WellFounded.sym2_gameAdd`: formalizes induction on unordered pairs, where exactly one entry
+decreases at a time.
 -/
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:36]: In the mathport version the line below this one uses `{}`

Comment (Parcly-Taxel) [reply Mathlib/Order/GameAdd.lean:35]: Both review comments addressed, @semorrison

File: Mathlib/Order/GameAdd.lean [hunk 3]
```diff
@@ -19,36 +19,68 @@ This file defines, given relations `rα : α → α → Prop` and `rβ : β →
...
 
 
-variable (rα : α → α → Prop) (rβ : β → β → Prop)
+variable {α β : Type _} {rα : α → α → Prop} {rβ : β → β → Prop}
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:37]: Lean3 has `/-! ### `prod.game_add` -/` here between the `variable` and `namespace` line

File: Mathlib/Order/GameAdd.lean [hunk 4]
```diff
@@ -19,36 +19,68 @@ This file defines, given relations `rα : α → α → Prop` and `rβ : β →
...
+#align prod.game_add_swap_swap Prod.gameAdd_swap_swap
+
+theorem gameAdd_swap_swap_mk (a₁ a₂ : α) (b₁ b₂ : β) :
+    GameAdd rα rβ (a₁, b₁) (a₂, b₂) ↔ GameAdd rβ rα (b₁, a₁) (b₂, a₂) :=
+  gameAdd_swap_swap rβ rα (b₁, a₁) (b₂, a₂)
+#align prod.game_add_swap_swap_mk Prod.gameAdd_swap_swap_mk
+
 /-- `GameAdd` is a subrelation of `Prod.Lex`. -/
 theorem gameAdd_le_lex : GameAdd rα rβ ≤ Prod.Lex rα rβ := fun _ _ h =>
   h.rec (Prod.Lex.left _ _) (Prod.Lex.right _)
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:86]: The docstring on line 89 was updated in Lean 3 too

Comment (Parcly-Taxel) [reply Mathlib/Order/GameAdd.lean:89]: :+1:

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3123
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3123~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/GameAdd.lean [hunk 5]
```diff
@@ -63,8 +98,6 @@ theorem rprod_le_transGen_gameAdd : Prod.RProd rα rβ ≤ Relation.TransGen (Ga
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:98]: Line 93 (on the right) should have a `Prod` removed too.

File: Mathlib/Order/GameAdd.lean [hunk 6]
```diff
@@ -84,3 +117,138 @@ theorem WellFounded.prod_gameAdd (hα : WellFounded rα) (hβ : WellFounded rβ)
...
+    WellFounded rα →
+      WellFounded rβ →
+        (∀ a₁ b₁, (∀ a₂ b₂, GameAdd rα rβ (a₂, b₂) (a₁, b₁) → C a₂ b₂) → C a₁ b₁) → ∀ a b, C a b :=
+  GameAdd.fix
+#align prod.game_add.induction Prod.GameAdd.induction
+
+end Prod
+
+/-! ### `Sym2.GameAdd` -/
+
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:150]:
```suggestion
```

**Review by eric-wieser — APPROVED** [review]

File: Mathlib/Order/GameAdd.lean [hunk 7]
```diff
@@ -89,7 +89,7 @@ theorem gameAdd_le_lex : GameAdd rα rβ ≤ Prod.Lex rα rβ := fun _ _ h =>
   h.rec (Prod.Lex.left _ _) (Prod.Lex.right _)
 #align prod.game_add_le_lex Prod.gameAdd_le_lex
 
-/-- `Prod.RProd` is a subrelation of the transitive closure of `GameAdd`. -/
+/-- `Prod.RProd` is a subrelation of the transitive closure of `Prod.GameAdd`. -/
 theorem rprod_le_transGen_gameAdd : Prod.RProd rα rβ ≤ Relation.TransGen (GameAdd rα rβ)
```
Comment (eric-wieser) [inline Mathlib/Order/GameAdd.lean:93]: The change I was referring to is
```suggestion
theorem rprod_le_transGen_gameAdd : RProd rα rβ ≤ Relation.TransGen (GameAdd rα rβ)
```
which is in the Lean 3 diff.

**Comment by semorrison** [comment]
Can we please have the links to the port status dashboard, to make it easier to review the mathlib3 changes?

**Comment by Parcly-Taxel** [comment]
> Can we please have the links to the port status dashboard, to make it easier to review the mathlib3 changes?

:white_check_mark:

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**